### PR TITLE
MAINT: work around arg{min,max} not implemented for booleans

### PR DESCRIPTION
**Pull request by ev-br**

### CODE REVIEW

File: torch_np/_detail/_reductions.py [hunk 1]
```diff
@@ -36,12 +36,22 @@ def count_nonzero(a, axis=None):
 
 def argmax(tensor, axis=None):
     axis = _util.allow_only_single_axis(axis)
+
+    if tensor.dtype == torch.bool:
+        # RuntimeError: "argmax_cpu" not implemented for 'Bool'
+        tensor = tensor.view(torch.int8)
```
Comment (ev-br) [inline torch_np/_detail/_reductions.py:42]: Is the equivalence guaranteed in pytorch?

Comment (lezcano) [reply torch_np/_detail/_reductions.py:42]: I would cast it to `uint8` to be 100% sure, but yeah, I believe it is guaranteed by some work @peterbell10 did some time ago.

**Comment by ev-br** [comment]
By cast you mean tensor.to or tensor.view?

**Comment by lezcano** [comment]
view is fine.

File: torch_np/_detail/_reductions.py (hunk 1 above)
Comment (peterbell10) [reply torch_np/_detail/_reductions.py:42]: A cast is guaranteed to be 0 or 1, but a view could contain any byte value.

Comment (peterbell10) [reply torch_np/_detail/_reductions.py:42]: Oh and of course the work I did doesn't apply to the `torch.compile` stack.

**Comment by ev-br** [comment]
Based on https://github.com/Quansight-Labs/numpy_pytorch_interop/pull/63#discussion_r1114856264 and https://github.com/Quansight-Labs/numpy_pytorch_interop/pull/63#discussion_r1114739469 let's copy to uint8 to be on the safe side.

File: torch_np/_detail/_reductions.py (hunk 1 above)
Comment (lezcano) [reply torch_np/_detail/_reductions.py]: In this case, if viewing with `uint8` we would get something that's non zero, so argmax would then work, but sure, let's do `to`, as the compiler should be able to optimise out the copy really

Comment (peterbell10) [reply torch_np/_detail/_reductions.py]: IIRC argmax always returns the lowest index that's equal to the maximum, so the distinction does matter.

**Pull request merged**